### PR TITLE
[Feature] Implement category and rules edit and some other small fixes

### PR DESCRIPTION
**Pull request by daiHash**
## Description
Implement category and rules edit and some other small fixes
Add labels to each input

## Closes issue(s)
#207 

## Screenshots
category-edit

## Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist

- [x] I have tested this code

## Other comments
Rules edit is left so I will work in the next PR

### CODE REVIEW

File: src/helpers/api/gardens/updateGardenData.ts [hunk 1]
```diff
@@ -3,8 +3,10 @@ import { api } from "../../../utils/api";
 
 export type GardenUpdateParams = {
   gardenId: string;
-  data: Partial<Pick<Garden, "name" | "description">> &
-    Pick<Garden, "fireBaseUserId" | "gardenCategoryId">;
+  data: Pick<
+    Garden,
+    "fireBaseUserId" | "gardenCategoryId" | "name" | "description"
+  >;
```
Comment (daiHash) [inline src/helpers/api/gardens/updateGardenData.ts:9]: Fix update api body

File: src/pages/Settings/components/CategorySelector/index.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          >
+            Update
+          </Button>
+          {updateGardenApi.status === "loading" && (
+            <CircularProgress size={28} className={classes.buttonProgress} />
+          )}
+        </div>
+      </div>
+    </>
+  );
```
Comment (daiHash) [inline src/pages/Settings/components/CategorySelector/index.tsx:172]: Added Selector component for Garden categories

File: src/pages/Settings/components/DescriptionInput/index.tsx [hunk 3]
```diff
@@ -65,20 +71,13 @@ export const DescriptionInput: React.FC<Props> = memo(
...
-          description: currentGardenDescription,
-          fireBaseUserId: garden.fireBaseUserId,
-          gardenCategoryId: garden.gardenCategoryId,
-        };
         updateGardenInfo({
-          gardenId: garden._id,
-          data,
+          gardenId,
+          data: { ...updateData, description: currentGardenDescription },
         });
```
Comment (daiHash) [inline src/pages/Settings/components/DescriptionInput/index.tsx:77]: Pass all the required data

File: src/pages/Settings/components/NameInput/index.tsx [hunk 4]
```diff
@@ -60,66 +66,79 @@ export const NameInput: React.FC<Props> = memo(
...
+                  disabled={disableUpdate}
+                >
+                  Update
+                </Button>
+                {updateGardenApi.status === "loading" && (
+                  <CircularProgress
+                    size={28}
+                    className={classes.buttonProgress}
+                  />
+                )}
```
Comment (daiHash) [inline src/pages/Settings/components/NameInput/index.tsx:134]: Right now theres a lot of similar code in each component I'd like to refactor later if we have time

File: src/pages/Settings/index.tsx [hunk 5]
```diff
@@ -83,6 +118,30 @@ export const MyNiwaSettings = () => {
...
+      fireBaseUserId: garden?.fireBaseUserId ?? "",
+      gardenCategoryId: initialGardenCategoryId,
+    }),
+    [
+      garden?.fireBaseUserId,
+      initialGardenCategoryId,
+      initialGardenDescription,
+      initialGardenName,
+    ]
+  );
```
Comment (daiHash) [inline src/pages/Settings/index.tsx:143]: Set all required data for garden update api body

File: src/pages/Settings/index.tsx [hunk 6]
```diff
@@ -103,37 +162,52 @@ export const MyNiwaSettings = () => {
...
+                  showSelector={showCategoryInput}
+                  gardenId={gardenId}
+                  updateData={updateGardenData}
+                  initialGardenCategoryId={initialGardenCategoryId}
+                  currentGardenCategoryId={currentGardenCategoryId}
+                  onCategoryInputChange={onCategoryInputChange}
+                  setShowCategoryInput={setShowCategoryInput}
+                  setGardenCategoryId={setGardenCategoryId}
+                  updateInitVal={setInitCategory}
                 />
```
Comment (daiHash) [inline src/pages/Settings/index.tsx:211]: would like to refactor this with some common component if we have time

**Review by kome12 — APPROVED** [review]
LGTM!

**Pull request merged**